File: src/temp/primitives/types/PBoolean.kt
package temp.primitives.types

import temp.primitives.Primitive

class PBoolean(var value: Boolean) : Primitive() {
    override fun getPrimitiveValue(): Any {
        return this.value
    }

    override fun negate(): Primitive {
        return PNone()
    }

    override fun not(): Primitive {
        return PBoolean(!this.value)
    }

    override fun plus(primitive: Primitive): Primitive {
        return PNone()
    }

    override fun minus(primitive: Primitive): Primitive {
        return PNone()
    }

    override fun times(primitive: Primitive): Primitive {
        return PNone()
    }

    override fun div(primitive: Primitive): Primitive {
        return PNone()
    }

    override fun rem(primitive: Primitive): Primitive {
        return PNone()
    }

    override fun compareGreater(primitive: Primitive): Primitive {
        return PNone()
    }

    override fun compareLesser(primitive: Primitive): Primitive {
        return PNone()
    }

    override fun toString(): String {
        return value.toString()
    }
}

File: src/codepulsar/primitive/type/PBoolean.cpp
#include "PBoolean.h"


Pulsar::PBoolean::PBoolean(bool value) {
    this->value = value;
}

Pulsar::PrimitiveType Pulsar::PBoolean::getPrimitiveType() {
    return PR_BOOLEAN;
}

bool Pulsar::PBoolean::isPrimitiveType(PrimitiveType primitiveType) {
    return primitiveType == PR_BOOLEAN;
}

std::any Pulsar::PBoolean::getPrimitiveValue() {
    return this->value;
}

Pulsar::Primitive* Pulsar::PBoolean::unaryNegate() {
    return new PNone();
}

Pulsar::Primitive* Pulsar::PBoolean::unaryNot() {
    return new PBoolean(!this->value);
}

Pulsar::Primitive* Pulsar::PBoolean::plus(Primitive* primitive) {
    return new PNone();
}

Pulsar::Primitive* Pulsar::PBoolean::minus(Primitive* primitive) {
    return new PNone();
}

Pulsar::Primitive* Pulsar::PBoolean::times(Primitive* primitive) {
    return new PNone();
}

Pulsar::Primitive* Pulsar::PBoolean::div(Primitive* primitive) {
    return new PNone();
}

Pulsar::Primitive* Pulsar::PBoolean::rem(Primitive* primitive) {
    return new PNone();
}

Pulsar::Primitive* Pulsar::PBoolean::compareGreater(Primitive* primitive) {
    return new PNone();
}

Pulsar::Primitive* Pulsar::PBoolean::compareLesser(Primitive* primitive) {
    return new PNone();
}

std::string Pulsar::PBoolean::toString() {
    return this->value ? "true" : "false";
}
